### PR TITLE
proposal for cli-commands api

### PR DESCRIPTION
**Pull request by memsharded**
Changelog: Feature: New ``CommandAPI`` subapi in the ``ConanAPI`` that allows calling other commands.
Docs: https://github.com/conan-io/docs/pull/3635

### CODE REVIEW

**Review by franramirez688 — COMMENTED** [review]
I think it looks really simple 👏

**Comment by jcar87** [comment]
I already have a few custom commands in mind if this is possible!! :D

**Review by czoido — APPROVED** [review]

File: conans/test/integration/command_v2/custom_commands_test.py [hunk 1]
```diff
@@ -286,3 +288,84 @@ def hello(conan_api, parser, *args, **kwargs):
...
+            import json
+            from conan.cli.command import conan_command
+            from conan.cli.formatters.graph import format_graph_json
+
+            @conan_command(group="custom commands", formatters={"json": format_graph_json})
+            def mycommand(conan_api, parser, *args, **kwargs):
+                \"""
+                mycommand help
+                \"""
+                result = conan_api.command.run("create", ".", "--version=1.0.0")
```
Comment (jcar87) [inline conans/test/integration/command_v2/custom_commands_test.py]: this fails if expressed this way, but passes if we call it with the arguments as a list
```
result = conan_api.command.run("create", [".", "--version=1.0.0"])
```

which seems to be a different syntax than the test above that passes them as separate arguments

**Review by jcar87 — APPROVED** [review]
LGTM!

**Review by AbrilRBS — APPROVED** [review]
Can't check right now because the question came to my mind just after I closed the laptop for the day, but I assume this works just fine for custom commands, right?

Otherwise looks good and feels nice to use! 👍

**Comment by memsharded** [comment]
> Can't check right now because the question came to my mind just after I closed the laptop for the day, but I assume this works just fine for custom commands, right?

@RubenRBS you mean that one custom command can also call other custom command? Yes, it should, I can add a test to cover this.

**Review by AbrilRBS — APPROVED** [review]

**Review by czoido — APPROVED** [review]

**Pull request merged**